### ISSUE TITLE
App doesn't inherit shell environment

### ISSUE THREAD

**Issue by joshaber** [desktop/desktop#1796]
The app doesn't inherit the shell environment. This causes some problems:

* Hooks don't work: https://github.com/desktop/desktop/issues/1734
* PGP signing won't work: https://github.com/desktop/desktop/issues/1391
* Possibly SSH shenanigans: https://github.com/desktop/desktop/issues/1795

Atom [does some stuff](https://github.com/atom/atom/pull/11054) to try to fix this. We should look into that.

**Comment by jjacode** [comment]
So what happened with this? It seems like it is still an issue. 2018-06-15

**Comment by cmer** [comment]
@jjacode I managed to fix issues similar to this for all my GUI apps by modifying `/etc/paths` (macOS). 

Some more complex things won't work, but this fixes anything path-related.

Here's what mine looks like:

```
/Users/carl/bin
/Users/carl/.rbenv/shims
/Users/carl/.rbenv/bin
/usr/local/bin
/usr/bin
/bin
/usr/sbin
/sbin
```

**Comment by shiftkey** [comment]
@jjacode #1815 should have resolved it. If you think you've stumbled upon something else feel free to open a fresh issue so we can understand what's happening.

**Comment by captDaylight** [comment]
I just installed the github app yesterday and I'm getting 

```
Commit failed - exit code 128 received, with output: 'fatal: cannot run gpg: No such file or directory
error: gpg failed to sign the data
fatal: failed to write commit object'
```

Does this have to do with this issue?

**Comment by shiftkey** [comment]
@eluchsinger @captDaylight if you'd like to provide some additional diagnostics related to #1815 you can run through these steps and see if your environment variables are setup correctly:

 - Open **View** | **Toggle Developer Tools** 
 - In the **Console** tab, enter `process.env.PATH` to see what it's currently set to

<img width="1227"  src="https://user-images.githubusercontent.com/359239/41974740-262bbc80-79ef-11e8-9b4b-88578d59c448.png">

That matches up with what's available in my shell, which is the intent of #1815. If they're different, let's figure out why by opening a fresh issue.
